cupy/cuda/memory.pyx

```diff
@@ -563,11 +564,13 @@ cdef class PooledMemory(BaseMemory):
 
         """
         cdef intptr_t ptr
+        cdef SingleDeviceMemoryPool pool
         ptr = self.ptr
         if ptr == 0:
             return
         self.ptr = 0
-        pool = self.pool()
+        pool = <SingleDeviceMemoryPool>weakref.PyWeakref_GetObject(self.pool)
```

Comment:
How much difference is there by using `weakref.PyWeakref_GetObject`?
`self.pool()` is easier to read.